### PR TITLE
Change and reset password

### PR DESCRIPTION
**Pull request by joe-yeager**
Added the endpoints for a user to change and reset their password.

### CODE REVIEW

**Comment by joe-yeager** [comment]
I just realized the reset password should probably have a lock time to prevent users from spamming themselves.  I'll add this tomorrow.

File: integration/controllers/user.ctrl.spec.ts [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+    it('should reset a password', (done) =>{
+        req.params.curPassword = req.params.newPassword;
+        req.params.email = user.email;
+        resetPassword(good.req)
+        .then(test200)
+        .then( () => {
+          changePassword(good.req)
+          .then(test401)
+          .then( () => {
+            req.params.curPassword = req.params.newPassword;
```
Comment (joe-yeager) [inline integration/controllers/user.ctrl.spec.ts:111]: This is in the wrong test, this needs to be in the change password test.

**Comment by joe-yeager** [comment]
#WeWereThere
This is a group peer, for those of you who weren't here.
Fixe the test and add the lock before merging these merge in develop before merging this.

**Comment by ianaldrighetti** [comment]
@joeYeager Is this ready to be merged? Looks like a merge is required on your end.

**Comment by joe-yeager** [comment]
Yeah, I was going to add a lockout for resetting a password, but if that's not a big deal, its ready.

**Pull request merged**